### PR TITLE
Automatically register for tx notifications after a rescan.

### PR DESCRIPTION
**Pull request by jrick**
This changes the behavior of the rescan RPC to automatically set the
client up for transaction notifiations for transactions paying to any
rescanned address and spending from the final UTXO set watched by the
rescan after a rescanned is performed through the best block in the
chain.

### CODE REVIEW

File: rpcwebsocket.go [hunk 1]
```diff
@@ -511,35 +511,37 @@ func (m *wsNotificationManager) notifyForNewTx(clients map[chan struct{}]*wsClie
 	}
 }
 
-// RegisterSpentRequest requests an notification when the passed outpoint is
-// confirmed spent (contained in a block connected to the main chain) for the
-// passed websocket client.  The request is automatically removed once the
-// notification has been sent.
-func (m *wsNotificationManager) RegisterSpentRequest(wsc *wsClient, op *wire.OutPoint) {
+// RegisterSpentRequests requests an notification when each of the passed
```
Comment (davecgh) [inline rpcwebsocket.go]: an -> a

**Comment by davecgh** [comment]
I haven't test this yet, but it reads well.  I'll test it shortly.

**Comment by jrick** [comment]
Test with btcsuite/btcwallet#199.  It removes the (now unnecessary) notification registrations at btcd connect.

**Comment by davecgh** [comment]
OK.  Please squash/rebase this and I'll get it merged.

**Pull request merged**